### PR TITLE
Fix babel transpilation & clean up grunt flow

### PR DESCRIPTION
**Pull request by parkershepherd**
## Description
A previous commit disabled babel, uglify wasn't being applied to the main map.js file, and the grunt flow was inconsistent (some files went to an intermediate `/css` folder, not all source maps were present, etc)

## Motivation and Context

* Makes it easier to keep track of which files are build vs source (not ALL built files are in `/dest`)
* Fixes babel
* Minifies map.js
* Makes sourcemaps work all the way down to original source

## How Has This Been Tested?
Chrome/OSX, Firefox/OSX, Safari/OSX, Chrome/Android

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by parkershepherd** [comment]
^ is travis broken? It's erroring on npm install, not on the actual test

**Comment by andrewda** [comment]
@parkershepherd happened on half of my commits, too. Seems like something's wrong with the config.

File: Gruntfile.js [hunk 1]
```diff
@@ -30,53 +33,52 @@ module.exports = function(grunt) {
       dist: {
         files: {
           'static/dist/js/app.js': 'static/js/app.js',
-          'static/dist/js/map.js': 'static/map.js'
+          'static/dist/js/map.js': 'static/map.js' // Todo: move map.js into the js folder
```
Comment (chuyskywalker) [inline Gruntfile.js]: Do eeet

**Comment by chuyskywalker** [comment]
So much +1 on this, but I've got some conflicts in #1721 that will conflict. Either way, is good.

**Comment by parkershepherd** [comment]
Haha yeah, I figured I'd clean it up after yours gets merged. (or before, don't care which)

**Comment by chuyskywalker** [comment]
#1721 is done in, rebase here and I'll get this in. Let's also move map.js too.

**Comment by parkershepherd** [comment]
Will do

**Comment by chuyskywalker** [comment]
I'm gonna get #2384 in as well tonight for even more cleanups

**Comment by parkershepherd** [comment]
As long as we're ready to ignore non-devs complaining the dev branch doesn't work :p

**Comment by chuyskywalker** [comment]
I might actually extend that PR to include a check and `exit()` in the application which says, "DID YOU NOT READ THE WIKI ABOUT GRUNT?"

**Comment by parkershepherd** [comment]
Alrighty, this should be good. You can nuke your `static/css` and `static/dist` folders, then run grunt. `static/dist` will get fully rebuilt (to test)

File: Gruntfile.js [hunk 2]
```diff
@@ -94,10 +105,10 @@ module.exports = function(grunt) {
   grunt.loadNpmTasks('grunt-html-validation');
   grunt.loadNpmTasks('grunt-babel');
 
-  grunt.registerTask('js-build', ['jshint', 'babel', 'uglify']);
+  grunt.registerTask('js-build', ['babel', 'uglify']);
   grunt.registerTask('css-build', ['sass', 'cssmin']);
 
-  grunt.registerTask('build', ['js-build', 'css-build']);
+  grunt.registerTask('build', ['jshint', 'clean', 'js-build', 'css-build']);
```
Comment (chuyskywalker) [inline Gruntfile.js]: Why not have `js-hint` be in `js-build` -- seems to be used in all those places together anyway, no?

Comment (parkershepherd) [inline Gruntfile.js]: Just wanted to give jshint a chance to complain about an error before nuking the `/dist` directory

Comment (parkershepherd) [inline Gruntfile.js]: Although leaving it in a broken state might be beneficial in case someone forgets to keep an eye on their grunt window?

Comment (chuyskywalker) [inline Gruntfile.js]: But, as far as I can tell, jshint doesn't get run against anything in dest, no? (I think I may just be confused about the order of operations here)

Comment (parkershepherd) [inline Gruntfile.js]: yeah, jshint just runs on pre-compiled code (`static/js`) but if jshint finds an error, it halts the build and throws the warning to the console. If the `clean` happened before `jshint`, and `jshint` happened to throw a warning, the dist folder would be empty. So `jshint` -> `nuke dist` -> `babel` -> `uglify`

**Comment by chuyskywalker** [comment]
The great dist clean out has been mereged, which has conflicted here. But I do want to get this in next.

**Comment by parkershepherd** [comment]
@chuyskywalker All up to date now

**Comment by chuyskywalker** [comment]
You missed a dist file: https://github.com/parkershepherd/PokemonGo-Map/tree/bb907f52ffe8b1506da64f0bbe904469cdb0accf/static/dist/js

**Comment by chuyskywalker** [comment]
Aside from that one file which was missed in the delete cleanup, this LGTM (tested and found no errors compiling, or using built js/css)

**Comment by parkershepherd** [comment]
@chuyskywalker nice catch. Should be gone now

**Pull request merged**